langchain/src/llms/cloudflare_workersai.ts

```diff
@@ -125,9 +125,69 @@ export class CloudflareWorkersAI
         (error as any).response = response;
         throw error;
       }
-      return response.json();
+      return response;
     });
+  }
 
-    return responseData.result.response;
+  async *_streamResponseChunks(
+    prompt: string,
+    options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<GenerationChunk> {
+    const response = await this._request(prompt, options, true);
+    if (!response.body) {
+      throw new Error("Empty response from Cloudflare. Please try again.");
+    }
+    const stream = convertEventStreamToIterableReadableDataStream(
+      response.body
+    );
+    for await (const chunk of stream) {
+      if (chunk !== "[DONE]") {
+        const parsedChunk = JSON.parse(chunk);
+        const generationChunk = new GenerationChunk({
+          text: parsedChunk.response,
+        });
+        yield generationChunk;
+        // eslint-disable-next-line no-void
+        void runManager?.handleLLMNewToken(generationChunk.text ?? "");
+      }
+    }
+  }
+
+  /** Call out to CloudflareWorkersAI's complete endpoint.
+   Args:
+       prompt: The prompt to pass into the model.
+   Returns:
+       The string generated by the model.
+   Example:
+   let response = CloudflareWorkersAI.call("Tell me a joke.");
+   */
+  async _call(
+    prompt: string,
+    options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): Promise<string> {
+    if (!this.streaming) {
+      const response = await this._request(prompt, options);
+
+      const responseData = await response.json();
+
+      return responseData.result.response;
+    } else {
+      const stream = await this._streamResponseChunks(
```

Comment:
```suggestion
      const stream = this._streamResponseChunks(
```
nit, unnecessary await